.buildkite/scripts/steps/publish_oas_docs.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+source .buildkite/scripts/common/util.sh
+
+echo "--- Publish OAS docs"
+
+deploy_to_bump() {
+  local file_path="${1:-}"
+  local doc_name="${2:-}"
+  local doc_token="${3:-}"
+  local branch="${4:-}"
+
+  echo "Checking diff for doc '$doc_name' against file '$file_path'..."
+  local result=$(bump diff $file_path --doc $doc_name --token $doc_token --branch $branch --format=json)
+  ## Bump.sh does not respond with JSON when the diff is empty so we need to handle possibly not JSON :'(
+  local change_count=$(tr '\n' ' '<<<$result | jq -R 'fromjson? | length')
+  if [[ ! -z $change_count && $change_count -gt 0 ]]; then
+    echo "Found $change_count changes..."
+    echo "About to deploy file '$file_path' to doc '$doc_name' on bump.sh..."
+    bump deploy $file_path \
+      --branch $branch \
+      --doc $doc_name \
+      --token $doc_token ;
+    echo ""
+    echo "Note: if there is a warning of unchanged docs we probably have unpublished deployments waiting."
+    echo "Go to https://bump.sh/elastic/dashboard to see all the docs in the hub."
+  else
+    echo "Did not detect changes for '$file_path'; not deploying. Got response: $result"
+  fi
+}
+
+if [[ "$BUILDKITE_BRANCH" == "main" ]]; then
+  BUMP_KIBANA_DOC_NAME="$(vault_get kibana-bump-sh kibana-doc-name)"
+  BUMP_KIBANA_DOC_TOKEN="$(vault_get kibana-bump-sh kibana-token)"
+  deploy_to_bump oas_docs/output/kibana.yaml $BUMP_KIBANA_DOC_NAME $BUMP_KIBANA_DOC_TOKEN main;
+fi
+
+if [[ "$BUILDKITE_BRANCH" == "8.x" ]]; then
```

Comment:
> if [[ "$BUILDKITE_BRANCH" == "8.x" ]]; then

Since we're currently not planning to publicly publish the future releases (other than main at this point), it seems like there are some options to consider:

A) Separate the diff and deploy actions and only do the diff (not deploy) for branches that aren't live yet, or
B) Deploy to an internal API document for branches that aren't live yet, or
C) Deploy to version-specific branches but (with manual release mode enabled) only release the "current" branch. Leave the other branches unreleased (assuming that still enables the diffs to successfully generate previews for all branches).

I will follow up with Bump.sh to see if there's a recommended option.

Reply:
Cool, thanks Lisa. FWIW I'm happy to keep 8.x internal, just saw there was already a branch created for it so figured pushing 8.x contents to it would be OK. As I understand, both `main` and `8x` live behind manual releases atm.

Reply:
It seems like it's feasible to have branches that we just never release publicly. So I've created an "8x-unreleased" branch in the Kibana API document (for the 8.x branch to deploy and generate previews against). And the 8.16 branch can go to "v8".